### PR TITLE
[core] add read batch size option

### PR DESCRIPTION
**Pull request by zhangjun0x01**
related to https://github.com/apache/incubator-paimon/issues/641

### CODE REVIEW

File: paimon-format/src/main/java/org/apache/paimon/format/orc/OrcFileFormat.java [hunk 1]
```diff
@@ -62,13 +61,15 @@ public class OrcFileFormat extends FileFormat {
     private final Properties orcProperties;
     private final org.apache.hadoop.conf.Configuration readerConf;
     private final org.apache.hadoop.conf.Configuration writerConf;
+    private final int readBatchSize;
 
-    public OrcFileFormat(Options formatOptions) {
+    public OrcFileFormat(Options formatOptions, int readBatchSize) {
```
Comment (zhuangchong) [inline paimon-format/src/main/java/org/apache/paimon/format/orc/OrcFileFormat.java:66]: I have a question? Why the `readBatchSize` attribute is not obtained from options.

Comment (zhangjun0x01) [reply paimon-format/src/main/java/org/apache/paimon/format/orc/OrcFileFormat.java:66]: Yes, we can get `readBatchSize` by `option.get('read. batch-size')`. But I don't think we should do this way. We should use `CoreOptions#READ_BATCH_SIZE.key()`, but the `paimon-format` module does not reference the `paimon-core` module, so I pass this parameter from `paimon-core` to `paimon-format`

File: paimon-common/src/main/java/org/apache/paimon/format/FileFormatFactory.java [hunk 2]
```diff
@@ -25,5 +25,5 @@ public interface FileFormatFactory {
 
     String identifier();
 
-    FileFormat create(Options formatOptions);
+    FileFormat create(Options formatOptions, int readBatchSize);
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/format/FileFormatFactory.java:28]: Maybe just create a class `FormatContext`:
```
FormatContext {
    Options formatOptions();
    int readBatchSize();
}
```

File: paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java [hunk 3]
```diff
@@ -77,18 +78,35 @@ public Optional<FileStatsExtractor> createStatsExtractor(RowType type) {
 
     /** Create a {@link FileFormat} from table options. */
     public static FileFormat fromTableOptions(
-            Options tableOptions, ConfigOption<String> formatOption) {
+            Options tableOptions,
+            ConfigOption<String> formatOption,
+            ConfigOption<Integer> readBatchSizeOption) {
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java:83]: Maybe we can remove this method.
Provide a util static method in `CoreOptions`.

File: paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java [hunk 4]
```diff
@@ -77,18 +78,35 @@ public Optional<FileStatsExtractor> createStatsExtractor(RowType type) {
...
     }
 
-    /** Create a {@link FileFormat} from format identifier and format options. */
+    @VisibleForTesting
     public static FileFormat fromIdentifier(String identifier, Options options) {
+        return fromIdentifier(identifier, 1000, options);
+    }
+
+    /** Create a {@link FileFormat} from format identifier and format options. */
+    public static FileFormat fromIdentifier(String identifier, int readBatchSize, Options options) {
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java:96]: fromIdentifier(String identifier, FormatContext context)

**Review by JingsongLi — COMMENTED** [review]
Thanks @zhangjun0x01 for the contribution, left some comments.

**Review by JingsongLi — COMMENTED** [review]
Thanks for the update. Last comment: Can you add a test for format?
For example, you can create a format with identifier `test`, and validate options in create like:
```
public class TestFileFormatFactory implements FileFormatFactory {

    @Override
    public String identifier() {
        return "test";
    }

    @Override
    public FileFormat create(FormatContext context) {
          assertThat(context.options().get(myKey)).isEqualTo(myValue);
          assertThat(context.batchSize()).isEqualTo(myBatchSize);
          throw new SuccessException();
    }
}

public static class SuccessException extends RutimeException {}
```

**Comment by JingsongLi** [comment]
I accidentally merged this PR.
@zhangjun0x01 You can add case in another PR, It is better to add case in `FileFormatTest`, test `CoreOptions.createFileFormat`.

**Comment by zhangjun0x01** [comment]
> I accidentally merged this PR. @zhangjun0x01 You can add case in another PR, It is better to add case in `FileFormatTest`, test `CoreOptions.createFileFormat`.

ok,I will add later

**Pull request merged**